embedded-graphics/MIGRATING-0.5-0.6.md

```diff
@@ -0,0 +1,248 @@
+The `icoord` and `ucoord` macros have been removed. Instead, use `Point::new(x, y)` and `Size::new(x, y)` respectively.
+
+`Point::zero()` and `Size::zero()` were also added to get zero-filled coordinates.
+
+```diff
+- let c = Coord::new(-10, 20);
++ let c = Point::new(-10, 20);
+
+- println!("X: {}, Y: {}", c.0, c.1);
++ println!("X: {}, Y: {}", c.x, c.y);
+
+- let p = UnsignedCoord::new(5, 15);
++ let p = Size::new(5, 15);
+
+- println!("X: {}, Y: {}", p.0, p.1);
++ println!("X: {}, Y: {}", p.width, p.height);
+```
+
+### Text
+
+Text is now drawn with the `Text` struct. A `TextStyle` must be provided which selects which font to draw the text with.
+
+```diff
+- use embedded_graphics::{
+-     prelude::*,
+-     fonts::Font6x8,
+- };
+-
+- display.draw(
+-     Font6x8::render_str("Hello Rust!")
+-         .translate(Coord::new(20, 30))
+-         .fill(Some(1u8))
+-         .stroke(Some(0u8)),
+- );
+
++ use embedded_graphics::{
++     fonts::{Font6x8, Text},
++     pixelcolor::BinaryColor,
++     prelude::*,
++     style::TextStyleBuilder,
++ };
++
++ // Create a new text style
++ let style = TextStyleBuilder::new(Font6x8)
++     .text_color(BinaryColor::On)
++     .background_color(BinaryColor::Off)
++     .build();
++
++ // Create a text at position (20, 30) and draw it using style defined above
++ Text::new("Hello Rust!", Point::new(20, 30))
++     .into_styled(style)
++     .draw(&mut display)?;
+```
+
+- A new `Text` struct is introduced. Instead of `YourFont::render_str("text")`, use `Text::new("text")`.
+- Text must be given a `TextStyle` for it to be drawn on a display. Create a style with `TextStyleBuilder` and add it with `.into_styled(style)`.
+- The chosen font is now part of the `TextStyleBuilder` creation process. Set it with `TextStyleBuilder::new(<your font here>)`.
+
+Macro usage has also changed:
+
+```diff
+- use embedded_graphics::{prelude::*, text_6x8};
+-
+- display.draw(
+-     text_6x8!(
+-         "Hello Rust!",
+-         stroke = Some(1u8.into()),
+-         fill = Some(0u8.into())
+-     )
+-     .translate(Coord::new(20, 30))
+- );
+
++ use embedded_graphics::{egtext, text_style, fonts::Font6x8, pixelcolor::BinaryColor, prelude::*};
++
++ egtext!(
++     text = "Hello Rust!",
++     top_left = (20, 30),
++     style = text_style!(
++         font = Font6x8,
++         text_color = BinaryColor::On,
++         background_color = BinaryColor::Off,
++     )
++ )
++ .draw(&mut display)?;
+```
+
+- All built in `text_*!()` macros are removed and replaced with the `egtext!()` macro.
+- `egtext!()` should be coupled with the `text_style!()` macro to create styled texts with a chosen font.
+
+### Primitives
+
+Primitives changes:
+
+- Note: The `Triangle` primitive was added. If you were using 3 separate lines, this primitive is suggested instead.
+
+- Primitive styles now default to a stroke width of `0` instead of `1`. If nothing is drawn to the display, check that you set the `stroke_width` in `PrimitiveStyleBuilder`.
```

Comment:
We need to mention that `PrimitiveStyle` is a replacement for the old `Style` and that drawing primitives now requires creating a `Styled`.